src/ucs/profile/profile_defs.h

```diff
@@ -137,6 +154,58 @@ void ucs_profile_global_cleanup();
  */
 void ucs_profile_dump();
 
+
+/*
+ * Start a range trace on an arbitrary event in a potentially nested fashion.
+ * A range tracing can be started in a function and potentially end in an
+ * another function or a recursive invocation of the same function.
+ *
+ * @param [in]     name        String name for the range.
+ * @param [in]     color       Color for the range.
+ * @param [inout]  id          Unique ID returned to stop tracing the event
+ *
+ * @return ID to be used to stop tracing a range
+ */
+void ucs_profile_range_start(const char *name, ucs_profile_color_t color,
```

Comment:
i would put all those as internal functions, used by scope_begin/end only in ucs_profile_record